pypeit/images/combineimage.py

```diff
@@ -4,133 +4,247 @@
-                # Mask
-                bitmask = imagebitmask.ImageBitMask()
-                mask_stack = np.zeros(shape, bitmask.minimum_dtype(asuint=True))
-            # Grab the lamp status
-            lampstat += [self.spectrograph.get_lamps_status(pypeitImage.rawheadlist)]
-            # Process
-            img_stack[kk,:,:] = pypeitImage.image
-            # Construct raw variance image and turn into inverse variance
-            if pypeitImage.ivar is not None:
-                ivar_stack[kk, :, :] = pypeitImage.ivar
-            else:
-                ivar_stack[kk, :, :] = 1.
-            # Mask cosmic rays
-            if pypeitImage.crmask is not None:
-                crmask_stack[kk, :, :] = pypeitImage.crmask
+                # Allocate arrays to collect data for each frame
+                shape = (self.nfiles,) + pypeitImage.shape
+                img_stack = np.zeros(shape, dtype=float)
+                scl_stack = np.ones(shape, dtype=float)
+#                ivar_stack= np.ones(shape, dtype=float)
+                rn2img_stack = np.zeros(shape, dtype=float)
+                procv_stack = np.zeros(shape, dtype=float)
+#                crmask_stack = np.zeros(shape, dtype=bool)
+                gpm_stack = np.zeros(shape, dtype=bool)
+                lampstat = [None]*self.nfiles
+                darkcurr = np.zeros(self.nfiles, dtype=float)
+                exptime = np.zeros(self.nfiles, dtype=float)
+
+            # Save the lamp status
+            lampstat[kk] = self.spectrograph.get_lamps_status(pypeitImage.rawheadlist)
+            # Save the dark current and exposure time
+            darkcurr[kk] = pypeitImage.detector['darkcurr']
+            exptime[kk] = pypeitImage.exptime
+            # Processed image
+            img_stack[kk] = pypeitImage.image
+#            # Construct raw variance image and turn into inverse variance
+#            if pypeitImage.ivar is not None:
+#                ivar_stack[kk] = pypeitImage.ivar
+#            # Mask cosmic rays
+#            if pypeitImage.crmask is not None:
+#                crmask_stack[kk] = pypeitImage.crmask
+            # Get the count scaling
+            if pypeitImage.img_scale is not None:
+                scl_stack[kk] = pypeitImage.img_scale
             # Read noise squared image
             if pypeitImage.rn2img is not None:
-                rn2img_stack[kk, :, :] = pypeitImage.rn2img
+                rn2img_stack[kk] = pypeitImage.rn2img * scl_stack[kk]**2
+            # Processing variance image
+            if pypeitImage.proc_var is not None:
+                procv_stack[kk] = pypeitImage.proc_var * scl_stack[kk]**2
             # Final mask for this image
-            # TODO This seems kludgy to me. Why not just pass ignore_saturation to process_one and ignore the saturation
-            # when the mask is actually built, rather than untoggling the bit here
+            # TODO: This seems kludgy to me. Why not just pass ignore_saturation
```

Comment:
I don't recall.  Maybe @jhennawi will..

Reply:
I think this is actually a comment from @jhennawi ...